src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -539,5 +539,10 @@ public static int Count<T>(this ImmutableArray<T> items, Func<T, bool> predicate
 
             return dictionary;
         }
+
+        internal static TSource AsSingleton<TSource>(this ImmutableArray<TSource> source)
+        {
+            return source.NullToEmpty().Length == 1 ? source[0] : default;
```

Comment:
Let's use `System.Linq.ImmutableArrayExtensions.Single()` (or `SingleOrDefault()` if the caller expects 0 or 1 elements) instead of creating a new method.

Reply:
Last time I checked all those would throw when the array has more than one element.

Reply:
Looks like we don't need `NullToEmpty()` call since `Length` won't throw on default. but `IList.Count` does.